nsqd/nsqd.go

```diff
@@ -370,6 +379,11 @@ func (n *NSQD) PersistMetadata() error {
 		return err
 	}
 
+	err = os.Remove(oldFileName)
```

Comment:
an idea: I don't know what it means for windows, but we could just drop a symlink in when we remove the old filename; that way i think we preserve rollback. We could then refuse to run if both are present (and one isn't a symlink) to make sure rolling forward again is clean. The symlink would mean most importantly we only update a single file going forward when persisting the metdata